### PR TITLE
implementation of ethercat connection and control commands

### PR DESCRIPTION
**Pull request by KishanSawant**

### CODE REVIEW

File: KELO_SDP/ConnectEthercat.c [hunk 1]
```diff
@@ -12,46 +12,178 @@
...
+  {
+    counter += 1;
+    rxpdo1_t msg;
+    msg.timestamp = time(NULL);
+    msg.command1 = COM1_ENABLE1 | COM1_ENABLE2 | COM1_MODE_VELOCITY;
+    msg.limit1_p = 20;
+    msg.limit1_n = -20;
+    msg.limit2_p = 20; 
+    msg.limit2_n = -20;
+    msg.setpoint1 = 2;
```
Comment (rahulsai1219) [inline KELO_SDP/ConnectEthercat.c]: Change both the setpoints to 3 and -3 respectively

**Review by rahulsai1219 — CHANGES_REQUESTED** [review]
Please do the changes according to the comments

File: KELO_SDP/ConnectEthercat.c (hunk 1 above)
Comment (KishanSawant) [reply KELO_SDP/ConnectEthercat.c]: done

**Review by rahulsai1219 — APPROVED** [review]

File: KELO_SDP/ConnectEthercat.c [hunk 2]
```diff
@@ -12,46 +12,178 @@
...
+    ecx_send_processdata(&ecx_context);
+
+    //  receiving response message
+    ecx_receive_processdata(&ecx_context, EC_TIMEOUTRET);
+
+    //  reading encoder pivot angle of all wheel units (radians)
+    for (unsigned int i = 0; i < nWheels; i++)
+    {
+      txpdo1_t *ecData = (txpdo1_t *)ecx_slave[index_to_EtherCAT[i]].inputs;
+      printf("Pivot angle of wheel(id) %d: %f \n", nWheels[i], ecData->encoder_pivot);
```
Comment (shankarkavya) [inline KELO_SDP/ConnectEthercat.c]: Use index_to_EtherCAT instead of nWheels

File: KELO_SDP/ConnectEthercat.c [hunk 3]
```diff
@@ -12,46 +12,178 @@
...
+  else
+  {
+    printf("Found %i number of slaves. \n", ecx_slavecount);
+  }
+  ecx_config_map_group(&ecx_context, IOmap, 0);
+
+  // reading all slave names w.r.t their id's
+  for (int i = 1; i <= ecx_slavecount; i++)
+  {
+    printf("slave(index) \t%i has name \t%s\n", i, ecx_slave[i].name);
```
Comment (shankarkavya) [inline KELO_SDP/ConnectEthercat.c]: Use index_to_EtherCAT instead of i

**Review by shankarkavya — CHANGES_REQUESTED** [review]

**Review by shankarkavya — APPROVED** [review]

**Pull request merged**